### ISSUE TITLE
Rename SourceName to SourceSpec

### ISSUE THREAD

**Issue by amarjandu** [DataBiosphere/azul#2843]
This was a FIXME that was part of a sketch done by @hannes-ucsc 
See https://github.com/DataBiosphere/azul/issues/2650#issue-771289746

```
Index: src/azul/indexer/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/azul/indexer/__init__.py b/src/azul/indexer/__init__.py
--- a/src/azul/indexer/__init__.py	(revision 46e23043c6a4421f588ce3ac9f17db488ae4c88a)
+++ b/src/azul/indexer/__init__.py	(date 1614202583703)
@@ -53,6 +53,9 @@
 SOURCE_NAME = TypeVar('SOURCE_NAME', bound='SourceName')
 
 
+# FIXME: Rename to SourceSpec/SOURCE_SPEC, and all .name to .spec
+#        <insert link to ticket here>
+
 class SourceName(ABC, Generic[SOURCE_NAME]):
     """
     The name of a repository source containing bundles to index. A repository
```

**Comment by amarjandu** [comment]
pushed this into in-progress during parking lot discussion w/ @hannes-ucsc .

This should be fixed before addressing #2750

**Comment by hannes-ucsc** [comment]
For demo, show source specifications in manifest and JSON responses, as well as a sampling of contributions and aggregate documents (using Kibana).
